Abilities/npc_dota_hero_skywrath_mage/skywrath_mage_concussive_shot.cs

```diff
@@ -39,5 +44,58 @@ protected override float RawDamage
                 return this.Ability.GetAbilitySpecialData("damage");
             }
         }
+
+        public Hero TargetHit
+        {
+            get
+            {
+                return EntityManager<Hero>.Entities.OrderBy(x => 
+                              x.Distance2D(this.Owner)).FirstOrDefault(x =>
+                                    x.IsAlive &&
+                                    x.IsVisible &&
+                                    !x.IsIllusion &&
+                                    x.IsValid &&
+                                    x.IsEnemy(Owner) &&
+                                    x.Distance2D(Owner) <= this.Radius);
+            }
+        }
+
+        public override float GetDamage(params Unit[] targets)
+        {
+            if (!targets.Any(x => x == this.TargetHit))
+            {
+                return 0;
+            }
+
+            var totalDamage = 0.0f;
+
+            var damage = this.RawDamage;
+            var amplify = this.Owner.GetSpellAmplification();
+            foreach (var target in targets)
```

Comment:
only use the TargetHit in this case instead of all given targets?

Reply:
okey